lib/redcord/migration/index.rb

```diff
@@ -5,6 +5,41 @@
 module Redcord::Migration::Index
   extend T::Sig
 
+  sig { params(model: T.class_of(Redcord::Base), index_name: Symbol).void }
+  def add_index(model, index_name)
+    if model.class_variable_get(:@@range_index_attributes).include?(index_name)
+      model.redis.sadd("#{model.model_key}:range_index_attrs", index_name.to_s)
+    elsif model.class_variable_get(:@@index_attributes).include?(index_name)
+      model.redis.sadd("#{model.model_key}:index_attrs", index_name.to_s)
+    else
+      raise(
+        Redcord::AttributeNotIndexed,
+        "#{index_name} is not an indexed attribute.",
+      )
+    end
+
+    # Loop through existing records and build the index
+    model.redis.scan_each(match: "#{model.model_key}:id:*") do |key|
+      index_val = model.redis.hmget(key, index_name.to_s).first
+      id = key.split(':').last.to_i
+
+      # Force refresh on the updated index attribute. Use multi to ensure
+      # consistency incase this migration failed in the middle
+      model.redis.multi
```

Comment:
Should you start the multi command before the `hmget` so that you ensure the value doesn't change? 

What are the reasons for using Multi/Exec vs a lua script?

Reply:
hmmm. yeah; it's possible that the index_val has changed when we call update_hash. I guess this has to be done via a lua script.